daemon/src/tokio_ext.rs

```diff
@@ -0,0 +1,14 @@
+use std::fmt;
+use std::future::Future;
+
+pub fn spawn_fallible<F, E>(future: F)
```

Comment:
interesting concept. Question on top - again unrelated to the PR - what was the motivation for spawning this task inside the oracle? isn't the whole sync triggered asynchronously?

Reply:
Parallelization.

The task cannot hold references meaning it cannot update the state once it is done, hence we do the loop inside the handler, spawn of a number of tasks which all send a message at the end. Thus, all the tasks are independent, one failing doesn't impact the others and we get to update the state only for the successful ones.